### PR TITLE
Allow empty UserVariables for string values

### PR DESCRIPTION
**Pull request by YcKe**
Allow empty UserVariables for string values

### CODE REVIEW

File: main/WebServer.cpp [hunk 1]
```diff
@@ -1691,7 +1691,7 @@ namespace http {
 			std::string variablename = request::findValue(&req, "vname");
 			std::string variablevalue = request::findValue(&req, "vvalue");
 			std::string variabletype = request::findValue(&req, "vtype");
-			if ((variablename == "") || (variablevalue == "") || (variabletype == ""))
+			if ((variablename == "") || (variabletype == "") || (variablevalue == "" && variabletype != "2"))
```
Comment (gizmocuz) [inline main/WebServer.cpp]: Thanks!

To keep it readable (also the below patch) could you change to code like:

if (
----(variablename == "") || 
----(variabletype == "") || 
----( (variablevalue == "") && (variabletype != "2") )
) {
   .....
}

**Comment by YcKe** [comment]
Sure, you want me also to replace the ".empty"with"== 0" to have it consistent in both functions?

**Comment by gizmocuz** [comment]
No, just to write all checks on different lines
.empty() is not the same as '0', looks ok like it is now ;)

**Comment by gizmocuz** [comment]
Thanks!

**Pull request merged**